framework/PostProcessors/BasicStatistics.py

```diff
@@ -1045,91 +800,80 @@ def getCovarianceSubset(desired):
-      #   the only reason to skip is if an invalid matrix is requested
-      #if percentile, special treatment
+    #
+    # percentile, this metric is handled differently
+    #
+    metric = 'percentile'
+    if len(needed[metric]['targets'])>0:
+      self.raiseADebug('Starting "'+metric+'"...')
+      for targetDict in self.toDo[metric]:
+        percent = float(targetDict['percent'])
+        prefix = targetDict['prefix'].strip()
+        for targetP in targetDict['targets']:
+          if pbPresent:
+            relWeight  = pbWeights['realization'] if targetP not in pbWeights['SampledVarsPbWeight']['SampledVarsPbWeight'].keys() else pbWeights['SampledVarsPbWeight']['SampledVarsPbWeight'][targetP]
+          varName = prefix + '_' + targetDict['percent'].strip() + '_' + targetP
+          outputDict[varName] = np.atleast_1d(self._computeWeightedPercentile(inputDict['targets'][targetP].values,relWeight,percent=float(percent)/100.0))
+
+    #collect only the requested calculations except percentile, since it has been already collected
+    #in the outputDict.
+    for metric, requestList  in self.toDo.items():
       if metric == 'percentile':
-        for pct,targets in params.items():
-          label = 'percentile_'+self.parameters['percentile_map'][pct]
-          outputDict[label] = dict((target,calculations[label][target]) for target in targets)
-      #if other scalar, just report the result
-      elif metric in self.scalarVals:
-        outputDict[metric] = dict((target,calculations[metric][target]) for target in params)
-      #if a matrix block, extract desired values
-      else:
-        if metric in ['pearson','covariance']:
-          outputDict[metric] = {}
-          for entry in params:
-            #check if it was skipped for some reason
-            if entry == self.skipped.get(metric,None):
-              self.raiseADebug('Metric',metric,'was skipped for parameters',entry,'!  See warnings for details.  Ignoring...')
-              continue
-            for target in entry['targets']:
-              if target not in outputDict[metric].keys():
-                outputDict[metric][target] = {}
-              targetIndex = calculations[metric]['params'].index(target)
-              for feature in entry['features']:
+        continue
+      for targetDict in requestList:
+        prefix = targetDict['prefix'].strip()
+        if metric in self.scalarVals:
+          for targetP in targetDict['targets']:
+            varName = prefix + '_' + targetP
+            outputDict[varName] = np.atleast_1d(calculations[metric][targetP])
+        #TODO someday we might need to expand the "skipped" check to include scalars, but for now
+        #   the only reason to skip is if an invalid matrix is requested
+        #if matrix block, extract desired values
+        else:
+          #check if it was skipped for some reason
+          skip = self.skipped.get(metric, None)
```

Comment:
Out of curiosity...when and why a metric can be skipped?

Reply:
for example, not variables,

Reply:
I honestly do not get it. Not enough variables? You mean dependent FOMs? (e.g. covariance for the VarianceBasedSensitivity matrix?)?

Reply:
Based on the code, for the matrix we compute, if there is missing targets or features, the matrix computation will be skipped.